shared/src/main/scala/pl/metastack/metarx.scala

```diff
@@ -46,4 +52,59 @@ package object metarx extends OptImplicits {
     def &&(rch: ReadChannel[Boolean]): ReadChannel[Boolean] = rch && boolVal
     def ||(rch: ReadChannel[Boolean]): ReadChannel[Boolean] = rch || boolVal
   }
+
+  implicit class ReadChannelNumericExtensions[T: Numeric](rch: ReadChannel[T])(implicit num: Numeric[T]) extends ReadChannelExtensionBase(rch) {
+    import num._
+
+    def +(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, plus)
+    def +(argument: T): ReadChannel[T] = TWithOperation(argument, plus)
+    def -(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, minus)
+    def -(argument: T): ReadChannel[T] = TWithOperation(argument, minus)
+    def *(other: ReadChannel[T]): ReadChannel[T] = otherWithOperation(other, times)
+    def *(argument: T): ReadChannel[T] = TWithOperation(argument, times)
+
+    def unary_-(other: ReadChannel[T]): ReadChannel[T] = rch.map(-_)
+    def abs(): ReadChannel[T] = rch.map(num.abs)
```

Comment:
Are you sure we want to add support for `abs()`? I believe, we should limit ourselves to operators and conversion functions such as `toInt`.